turbo_audio/src/plugins/audio_api.rs

```diff
@@ -43,6 +43,22 @@ pub fn create_audio_api(fft_result: Arc<RwLock<FftResult>>) -> AudioApi {
         fft_result.read().unwrap().get_max_frequency()
     }
 
+    extern "C" fn get_bin_count(instance: *const std::ffi::c_void) -> std::ffi::c_ulonglong {
+        let fft_result = unsafe { &*(instance as *const Arc<RwLock<FftResult>>) };
+        fft_result.read().unwrap().get_bin_count() as _
+    }
+
+    extern "C" fn get_bin_value_at_index(
+        instance: *const std::ffi::c_void,
+        index: std::ffi::c_ulonglong,
+    ) -> std::ffi::c_float {
+        let fft_result = unsafe { &*(instance as *const Arc<RwLock<FftResult>>) };
+        fft_result
+            .read()
```

Comment:
This should never block right? Since we only write to this when updating the FFT result (which is made before we tick plugins). Or is there something I am missing? If this does lock it would also lock the core loop right?